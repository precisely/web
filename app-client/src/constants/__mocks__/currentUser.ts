function mockCurrentUser() {
  return {
    signup: jest.fn(),
    __mockSignupSuccessCase() {
      currentUser.signup.mockImplementationOnce(
        (email: string, password: string, onSuccess: () => void) => {
          onSuccess();
        }
      );
    },
    __mockSignupFailureCase() {
      currentUser.signup.mockImplementationOnce(
        (
          email: string,
          password: string,
          onSuccess: () => void,
          onFailure: (error: Error) => void
        ) => {
          onFailure(new Error('Unable to signup'));
        }
      );
    },

    isAuthenticated: jest.fn(),
    __mockisAuthenticatedSuccessCase() {
      currentUser.isAuthenticated.mockReturnValueOnce(true);
    },
    __mockisAuthenticatedFailureCase() {
      currentUser.isAuthenticated.mockReturnValueOnce(false);
    },

    logout: jest.fn(),
    __mockLogoutSuccessCase() {
      currentUser.logout.mockReturnValueOnce(true);
    },
    __mockLogoutFailureCase() {
      currentUser.logout.mockReturnValueOnce(false);
    },

    showLogin: jest.fn(),
    __mockshowLoginSuccessCase() {
      currentUser.showLogin.mockReturnValueOnce(true);
    },
    __mockshowLoginFailureCase() {
      currentUser.showLogin.mockReturnValueOnce(false);
    },

    resetPassword: jest.fn(),
    __mockResetPasswordSuccessCase() {
      currentUser.resetPassword.mockImplementationOnce(
        (
          email: string,
          verificationCode: string,
          newPassword: string,
          successCallback?: () => void
        ) => {
          successCallback();
        }
      );
    },
    __mockResetPasswordFailureCase() {
      currentUser.resetPassword.mockImplementationOnce(
        (
          email: string,
          verificationCode: string,
          newPassword: string,
          successCallback?: () => void,
          failureCallback?: (error: Error) => void
        ) => {
          failureCallback(new Error('Unable to reset the password.'));
        }
      );
    },

    __resetCurrentUserMocks: () => (currentUser = mockCurrentUser())
  };
}

export let currentUser = mockCurrentUser();
