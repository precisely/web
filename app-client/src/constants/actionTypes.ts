/*
 * Copyright (c) 2017-Present, CauseCode Technologies Pvt Ltd, India.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or
 * without modification, are not permitted.
 */

// All the constants for action types should be written here
export const FETCH_DATA: string = 'containers/Demo/FETCH_DATA';

export const SET_LENGTH: string = 'containers/Demo/SET_LENGTH';
