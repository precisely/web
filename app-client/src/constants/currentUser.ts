import {AWSUser} from 'src/utils/AWSUser.ts';

// tslint:disable-next-line
export const currentUser = new AWSUser();
