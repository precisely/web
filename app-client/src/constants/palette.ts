/*
 * Copyright (c) 2017-Present, CauseCode Technologies Pvt Ltd, India.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or
 * without modification, are not permitted.
 */

// All the global CSS styles and color constants should be written here
