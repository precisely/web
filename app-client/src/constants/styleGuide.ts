/*
 * Copyright (c) 2017-Present, Precise.ly, Inc.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or
 * without modification, are not permitted.
 */

// All the global CSS styles and color constants should be written here

import {CSS} from 'src/interfaces';

export const buttonDefault: string = '#6c757d';
export const buttonHover: string = '#545b62';
export const defaultBackground: string = '#F5F5F5';

export const helveticaFont: CSS = {
  fontFamily: 'Helvetica Neue, Helvetica, Arial, sans-serif'
};

export const inputStyle: CSS = {
  width: '385px',
  height: '44px',
  '@media screen and (max-width: 700px)': {
    width: '249px',
  },
  ':focus': {
    borderColor: '#d9d9d9',
  }
};

export const formButton: CSS = {
  width: '100%',
  color: '#fff',
  backgroundColor: buttonDefault,
  borderColor: buttonDefault,
  marginTop: '15px',
  ...inputStyle,
  ':hover': {
    backgroundColor: buttonHover,
    borderColor: buttonHover,
  }
};

export const removeBorderRadius: CSS = {
  borderRadius: 0,
};

export const noBorderTop: CSS = {
  borderTop: 'none',
  ...removeBorderRadius,
};

export const container: CSS = {
  height: '-webkit-fill-available',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: defaultBackground,
  padding: 0,
};

export const header: CSS = {
  textAlign: 'center',
  fontWeight: 200,
  fontSize: '30px',
  marginBottom: '25px',
  marginTop: '65px',
};

export const loginAndSignupPanel: CSS = {
  width: '597px',
  '@media screen and (max-width: 700px)': {
    width: '345px',
  },
};

export const alignCenter: CSS = {
  textAlign: '-webkit-center',
};

export const formMargin: CSS = {
  marginTop: '37px',
  marginBottom: '35px',
};
