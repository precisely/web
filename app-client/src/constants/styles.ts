/*
 * Copyright (c) 2011-Present, Precise.ly, Inc.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or
 * without modification, are not permitted.
 */


import { CSSProperties } from 'react';


export type ExtendedCSSProperties = CSSProperties & { [key: string]: any }; // to accomodate media queries and other special Radium strings


export const colors: {[key: string]: string} = {
  grey1: '#6c757d',
  grey2: '#545b62',
  white: '#FFF',
  offWhite: '#F5F5F5',
  blue: '#2b3fe0',
  disabledBlue: 'rgb(202, 208, 246)',
  preciselyOrange: '#FF8A4B',
  preciselyMagenta: '#c83a6e',
  preciselyGreen: '#00bc3e',
  preciselyPurple: '#a56aa3',
  preciselyPurpleAlpha: 'rgba(165, 106, 163, 0.85)',
  defaultTextColor: '#4a4a4a',
  defaultBackground: '#F5F5F5'
};


export const analysisColors: {[key: string]: string} = {
  defective: '#fc3f28',
  normal: '#00bc3e',
  enhanced: '#2B3FE0',
  unknown: '#9B9B9B'
};


export const fonts: {[key: string]: CSSProperties} = {

  helvetica: {
    fontFamily: 'Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif'
  },

  helveticaThin: {
    fontFamily: 'HelveticaNeue-Light, Helvetica Neue Light, Helvetica Neue, Helvetica, Arial, Lucida Grande, sans-serif',
    fontWeight: 300
  }

};


export const actionButtonStyle: CSSProperties = {
  backgroundColor: 'rgba(43, 63, 224, 0.8)',
  borderRadius: '8px',
  border: 0,
  paddingLeft: '85px',
  paddingRight: '85px',
  paddingTop: '5px',
  paddingBottom: '5px',
  fontSize: '16px',
  fontWeight: 300,
  cursor: 'pointer',
  color: colors.white
};
