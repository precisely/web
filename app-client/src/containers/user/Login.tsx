/*
 * Copyright (c) 2011-Present, Precise.ly, Inc.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or
 * without modification, are not permitted.
 */

import * as React from 'react';
import * as Radium from 'radium';
import {RouteComponentProps} from 'react-router';
import {Button, Form, FormGroup, Input, Link, InputGroupAddon, InputGroup} from 'src/components/ReusableComponents';
import {CSS} from 'src/interfaces';
import {PageContent} from 'src/components/PageContent';
import {login} from 'src/utils/cognito';
import {validateEmailAndPassword, showAlert} from 'src/utils';
import {NavigationBar} from 'src/components/navigationBar/NavigationBar';
import {
    formButton,
    removeBorderRadius,
    noBorderTop,
    loginAndSignupPanel,
    header,
    inputStyle,
    alignCenter,
    formMargin,
} from 'src/constants/styleGuide';

export interface ILoginState {
    email?: string;
    password?: string;
    isLoading?: boolean;
}

@Radium
export class Login extends React.Component<RouteComponentProps<void>, ILoginState> {

    toastId: number = null;

    state: ILoginState = {email: '', password: '', isLoading: false};

    updateLoadingState = (isLoading: boolean): void => {
        this.setState({isLoading});
    }

    onSuccess = (): void => {
        this.updateLoadingState(false);
        this.props.history.push('/dashboard');
    }

    onFailure = (message: string = 'Unable to login.'): void => {
        this.updateLoadingState(false);
        this.toastId = showAlert(this.toastId, message);
    }

    submitForm = (e: React.FormEvent<HTMLFormElement>): void => {
        e.preventDefault();
        const {email, password} = this.state;

        const validationInfo: {isValid: boolean, toastId: number} =
                validateEmailAndPassword(email, password, this.toastId);

        // This is needed to prevent multiple toast from getting rendered.
        this.toastId = validationInfo.toastId;

        if (validationInfo.isValid) {
            this.updateLoadingState(true);
            login(email, password, this.onSuccess, this.onFailure);
        }
    }

    handleInputChange(inputType: string, value: string): void {
        this.setState((): ILoginState => ({
            [inputType]: value,
        }));
    }

    render(): JSX.Element {
        const {isLoading, email, password} = this.state;

        return (
            <div>
                <NavigationBar {...this.props} />
                <div className="mx-auto" style={loginAndSignupPanel}>
                    <h3 style={header}>Welcome back</h3>
                    <PageContent>
                        <Form id="loginForm" onSubmit={this.submitForm} style={formMargin}>
                            <FormGroup className="mb-0" style={alignCenter}>
                                <Input
                                        style={[removeBorderRadius, inputStyle]}
                                        type="email"
                                        id="email"
                                        placeholder="Email"
                                        value={email}
                                        onChange={(e: React.ChangeEvent<HTMLInputElement>): void => {
                                            this.handleInputChange(e.target.id, e.target.value);
                                        }}
                                />
                            </FormGroup>
                            <FormGroup style={alignCenter}>
                                <InputGroup style={inputStyle}>
                                    <Input
                                            style={passwordStyle}
                                            type="password"
                                            id="password"
                                            placeholder="Password"
                                            value={password}
                                            onChange={(e: React.ChangeEvent<HTMLInputElement>): void => {
                                                this.handleInputChange(e.target.id, e.target.value);
                                            }}
                                    />
                                    <InputGroupAddon style={inputAddon}>
                                        <Link style={linkFontSize} to="/forgot-password">Forgot Password?</Link>
                                    </InputGroupAddon>
                                </InputGroup>
                            </FormGroup>
                            <Button style={formButton} disabled={isLoading} active={isLoading}>
                                {isLoading ? 'Please wait...' : 'Login'}
                            </Button>
                        </Form>
                    </PageContent>
                    <div style={[linkFontSize, {textAlign: 'center'}]} className="mt-4">
                        Don't have account? <Link to="/signup">Sign Up</Link>
                    </div>
                </div>
            </div>
        );
    }
}

const linkFontSize: CSS = {
    fontSize: '14px',
};

const passwordStyle: CSS = {
    ...noBorderTop,
    borderRight: 'none',
    ':focus': {
        borderColor: '#d9d9d9',
    }
};

const inputAddon: CSS = {
    ...noBorderTop,
    backgroundColor: '#fff',
};
