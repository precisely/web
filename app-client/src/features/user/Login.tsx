/*
 * Copyright (c) 2011-Present, Precise.ly, Inc.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or
 * without modification, are not permitted.
 */
import Auth0Lock from 'auth0-lock';
import {Component} from 'react';
import {authLockButtonBackground} from 'src/constants/styleGuide';
import { RouteComponentProps} from 'react-router';
import * as AuthUtils from 'src/utils/auth';

const LOGO = require('src/assets/logo.png');
const KEY_FOR_LAST_PATH_BEFORE_LOGIN = 'lastPathBeforeLogin';

export class Login extends Component<RouteComponentProps<void>> {
  lock = new Auth0Lock(
    process.env.REACT_APP_AUTH0_CLIENT_ID,
    process.env.REACT_APP_AUTH0_TENANT,
    {
      theme: {
        logo: LOGO,
        primaryColor: authLockButtonBackground
      },
      languageDictionary: {
        title: 'Precise.ly'
      },
      auth: {
        responseType: 'token id_token',
        params: {
          scope: 'openid profile'
        }
      }
    }
  );

  // tslint:disable-next-line
  constructor(props: any) {
    super(props);
    this.lock.on('authenticated', this.onAuthentication);
    this.lock.on('authorization_error', (error: Error) => {
      // TODO - Need to add error reporting
      // tslint:disable-next-line
      console.log('something went wrong', error.message);
    });
  }

  onAuthentication = (authResult: AuthResult) => {
    this.saveAuthenticationInfo(authResult);
    this.redirectAfterLogin();
  }

  redirectAfterLogin() {
    const lastPath = this.getLastPathBeforeLogin();
    this.saveLastPath(null);
    this.props.history.push(lastPath);
  }

  saveAuthenticationInfo (authResult: AuthResult) {
    AuthUtils.saveToken(authResult.accessToken, authResult.expiresIn);
  }

  saveLastPath (path: string | null) {
    if (path) {
      localStorage.setItem(KEY_FOR_LAST_PATH_BEFORE_LOGIN, path);
    }
    localStorage.removeItem(KEY_FOR_LAST_PATH_BEFORE_LOGIN);
  }
  
  getLastPathBeforeLogin() {
    const lastPage = localStorage.getItem(KEY_FOR_LAST_PATH_BEFORE_LOGIN);
    return lastPage ? lastPage : '/';
  }

  render(): JSX.Element {
    // Avoid showing Lock when hash is parsed.
    if (!(/access_token|id_token|error/.test(this.props.location.hash)) && !AuthUtils.isAuthenticated()) {
      if (this.props.location.state) {
        this.saveLastPath(this.props.location.state.from);
      }
      this.lock.show();
    } 
    return null;
  }
}
