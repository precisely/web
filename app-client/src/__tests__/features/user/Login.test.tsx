/*
 * Copyright (c) 2011-Present, Precise.ly, Inc.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or
 * without modification, are not permitted.
 */

jest.mock('src/constants/currentUser');

import * as React from 'react';
import * as Adapter from 'enzyme-adapter-react-16';
import * as Radium from 'radium';
import * as renderer from 'react-test-renderer';
import {RouteComponentProps} from 'react-router';
import {configure,shallow,ShallowWrapper} from 'enzyme';
import {Login} from 'src/features/user/Login';
import {currentUser} from 'src/constants/currentUser';
import {
  mockedHistory,
  mockedMatch,
  mockedLocation
} from 'src/__tests__/testSetup';

Radium.TestMode.enable();

configure({ adapter: new Adapter() });

describe('Login tests After Logging Out :', () => {
    currentUser[`__mockisAuthenticatedFailureCase`]();
    const componentTree: ShallowWrapper<RouteComponentProps<void>> = shallow(
        <Login
          history={mockedHistory}
          match={mockedMatch()}
          location={mockedLocation}
          staticContext={{}}
        />
    );

    it('should call showLogin method if user is unauthenticated : ', () => {
      expect(currentUser.showLogin).toBeCalled();
    });
});

describe('Login tests After Logging In :', () => {
  currentUser[`__mockisAuthenticatedSuccessCase`]();
  const componentTree: ShallowWrapper<RouteComponentProps<void>> = shallow(
      <Login
        history={mockedHistory}
        match={mockedMatch()}
        location={mockedLocation}
        staticContext={{}}
      />
  );

  it('should call push method on mockedHistory if user is authenticated : ', () => {
    expect(mockedHistory.push).toBeCalled();
  });

  it('should not render the Login component if user is authenticated : ', () => {
    expect(componentTree.find(Login).length).toBe(0);
  });
});

describe('Login tests Snapshot Testing :', () => {
    it('renders correctly', () => {
        const tree = renderer
          .create(
            <Login
              history={mockedHistory}
              match={mockedMatch()}
              location={mockedLocation}
              staticContext={{}}
            />
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
