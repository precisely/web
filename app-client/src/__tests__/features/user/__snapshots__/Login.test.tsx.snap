// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login tests Snapshot Testing : renders correctly 1`] = `null`;
